### ISSUE TITLE
Alexa middleware verifier doesn't accept google request

### ISSUE THREAD

**Issue by gwythyr** [jovotech/jovo-framework-nodejs#257]
## I'm submitting a...

- [x ] Bug report
- [ ] Feature request
- [ ] Documentation issue or request
- [ ] Other... Please describe:

### Error log
```
If you have an error log, please paste it here.
```

## Your Environment

* Jovo Framework version used: 1.x.x
* Operating System:

**Comment by aswetlow** [comment]
Good point. Could you try this example? (not tested)
```
const verifier = require('alexa-verifier-middleware');
let router = express.Router();
router.use('/webhook_alexa', verifier);
Webhook.use(router);


Webhook.listen(port, () => {
    console.log(`Example server listening on port ${port}!`);
});
Webhook.post('/webhook_alexa', (req, res) => {
    app.handleWebhook(req, res);
});
Webhook.post('/webhook', (req, res) => {
    app.handleWebhook(req, res);
});
```

**Comment by gwythyr** [comment]
@aswetlow i tested your code with both with Alexa and Assistant and found out that solution was failing with Alexa. Here's a response:
```
{
    "status": "failure",
    "reason": "The raw request body has already been parsed."
}
```
As i see reason is next: bodyParser is attached to express as middleware (jovo-framework/index.js:15) BEFORE attaching verifier middleware in index.js. So, for solving this without changing code of Jovo framework we should somehow remove bodyParser middleware, add verifier and add bodyParser again. I don't know is it possible? Maybe there are other way?

**Comment by aswetlow** [comment]
Yeah, you're right.  I found this: https://github.com/patrick-steele-idem/express-resetter

Last possibility without changing the framework code.
You don't import the Jovo-Webhook module, you create your express object on your own.

```
const verifier = require('alexa-verifier-middleware');

let server = express();
server.use('/webhook', bodyParser.json());

let router = express.Router();
router.use('/webhook_alexa', verifier);
router.use('/webhook_alexa', bodyParser.json());
Webhook.use(router);

// check for running ngrok tunnel
server.listen = function listen() {

    let server = http.createServer(this);
    return server.listen.apply(server, arguments); // eslint-disable-line
};

Webhook.listen(port, () => {
    console.log(`Example server listening on port ${port}!`);
});
Webhook.post('/webhook_alexa', (req, res) => {
    app.handleWebhook(req, res);
});
Webhook.post('/webhook', (req, res) => {
    app.handleWebhook(req, res);
});
```
(not tested)

**Comment by gwythyr** [comment]
@aswetlow finally, i've got working solution, but wasn't success in running your last code example. Here's working solution (file index.js): 
```
const {app} = require('./app/app.js');
const express = require('express');
const verifier = require('alexa-verifier-middleware');
const bodyParser = require('body-parser');
const http = require('http');

const Webhook = express();
Webhook.listen = function() {
    try {
        const verifier = require('alexa-verifier-middleware');
        let router = express.Router(); //eslint-disable-line
        Webhook.use(router);
        router.use('/webhook_alexa', verifier);
        router.use('/webhook_alexa', bodyParser.json());
        router.use('/webhook', bodyParser.json());
        let server = http.createServer(this);
        return server.listen.apply(server, arguments); // eslint-disable-line
    } catch (error) {
        if (error.code === 'MODULE_NOT_FOUND') {
            console.log('  Please install module alexa-verifier-middleware');
        } else {
            console.log(error);
        }
    }
};

if (app.isWebhook()) {
    const port = process.env.PORT || 10000;
    Webhook.listen(port, () => {
        console.log(`Example server listening on port ${port}!`);
    });
    Webhook.post('/webhook_alexa', (req, res) => {
        app.handleWebhook(req, res);
    });
    Webhook.post('/webhook', (req, res) => {
        app.handleWebhook(req, res);
    });
}
```
Would you consider to enable this functional into Jovo framework?

**Comment by jankoenig** [comment]
Hi @gwythyr , thanks, we're taking this into account for `v2`. Glad you got it working

**Comment by mfazul** [comment]
> @aswetlow finally, i've got working solution and also i wasn't successfull in running your last code example. Here's my working solution (file index.js):
> 
> ```
> const {app} = require('./app/app.js');
> const express = require('express');
> const verifier = require('alexa-verifier-middleware');
> const bodyParser = require('body-parser');
> const http = require('http');
> 
> const Webhook = express();
> Webhook.listen = function() {
>     try {
>         let router = express.Router(); //eslint-disable-line
>         Webhook.use(router);
>         router.use('/webhook_alexa', verifier);
>         router.use('/webhook_alexa', bodyParser.json());
>         router.use('/webhook', bodyParser.json());
>         let server = http.createServer(this);
>         return server.listen.apply(server, arguments); // eslint-disable-line
>     } catch (error) {
>         if (error.code === 'MODULE_NOT_FOUND') {
>             console.log('  Please install module alexa-verifier-middleware');
>         } else {
>             console.log(error);
>         }
>     }
> };
> 
> if (app.isWebhook()) {
>     const port = process.env.PORT || 10000;
>     Webhook.listen(port, () => {
>         console.log(`Example server listening on port ${port}!`);
>     });
>     Webhook.post('/webhook_alexa', (req, res) => {
>         app.handleWebhook(req, res);
>     });
>     Webhook.post('/webhook', (req, res) => {
>         app.handleWebhook(req, res);
>     });
> }
> 
> exports.handler = (event, context, callback) => {
>     app.handleLambda(event, context, callback);
> };
> ```
> 
> Would you consider somehow to enable this functional into Jovo framework? Or give me a hint how to do this enabling.

Can you provide the app.js file also?

**Comment by gwythyr** [comment]
There was no additional changes to app.js

**Comment by mfazul** [comment]
> There was no additional changes to app.js

But I am getting error that "app.handleWebhook" is not defined in app.js.

**Comment by gwythyr** [comment]
```
const {ExpressJS} = require('jovo-framework');
const verifier = require('alexa-verifier-middleware');
const bodyParser = require('body-parser');
const {app} = require('./app.js');
const express = require('express');
const Webhook = express();

if (process.argv.indexOf('--webhook') > -1) {
    const port = process.env.JOVO_PORT || 3000;
    Webhook.jovoApp = app;

    Webhook.listen(port, () => {
        console.info(`Local server listening on port ${port}.`);
    });

    Webhook.post('/alexa/', verifier, bodyParser.json(), async (req, res) => {
        await app.handle(new ExpressJS(req, res));
    });

    Webhook.post('/assistant/', bodyParser.json(), async (req, res) => {
        await app.handle(new ExpressJS(req, res));
    });
}
```

**Comment by mfazul** [comment]
I am using google cloud function in Alexa skill endpoint and getting following errors while going for the skill certification.
The skill end-point is not validating the signatures for incoming requests and is accepting requests with an empty signature URL.
The skill end-point is not validating the signatures for incoming requests and is accepting requests with an incorrect certificate URL.
The skill end-point is not validating the signatures for incoming requests and is accepting requests with an invalid signature URL specified.
The skill end-point is not validating the signatures for incoming requests and is accepting requests when no signature URL headers are specified.


Need a fix for this issue, can you please help me out to fix it.
